06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,221 @@
+          try {
+            InvertedIndex.validateFile(fileContent, fileName);
+            const content = JSON.parse(fileContent);
+            tranFile.push(fileName);
+            tranFile.push(content);
+            resolve(tranFile);
+          } catch (e) {
+            reject(e);
+          }
+        };
+      })(currentFile);
+      bookReader.readAsText(currentFile);
+    });
+  }
+  /**
+   * @validateFile method
+   * @param {fileContent} fileContent
+   * @param {fileName} fileName
+   * @returns {Boolean}
+   * reads the content of the book
+   */
+  static validateFile(fileContent, fileName) {
+    const fileExt = fileName.split('.').pop();
+    if (fileExt !== 'json') {
+      const error = `${fileName} has an Invalid File extension, JSON only`;
+      throw new Error(error);
+    }
+    let content;
+    try {
+      content = JSON.parse(fileContent);
+    } catch (e) {
+      const error = `OOPS!!! ${fileName} is not well formatted`;
+      throw new Error(error);
+    }
+    if (content.length === 0) {
+      const error = `${fileName} is an empty JSON file`;
+      throw new Error(error);
+    }
+    content.forEach((elem) => {
+      if (!elem.title || !elem.text) {
+        const error = `OOPS!!! ${fileName} does not contain title and text`;
+        throw new Error(error);
+      }
+    });
+    return true;
+  }
+
+  /**
+   * @searchIndex method
+   * @param {keyword} keyword
+   * @param {locations} locations
+   * @returns {Boolean}
+   * reads the content of the book
+   */
+  searchIndex(keyword, locations) {
+    const books = Object.keys(this.indicies);
+    if (!keyword) {
+      const error = 'please enter a keyword to search.';
+      throw new Error(error);
+    }
+
+    this.finalResult = {};
+    if (!locations || books.length === 0) {
+      const error = 'No file has been indexed yet';
+      throw new Error(error);
+    } else {
+      locations = locations || Object.keys(this.indicies);
+    }
+    locations.forEach((fileName) => {
+      const result = this.getResult(keyword, fileName);
+      this.finalResult[fileName] = result;
+    });
+    return true;
+  }
+
+  /**
+   * @getResult method
+   * @param {keyword} keyword
+   * @param {fileName} fileName
+   * @returns {Array}
+   * get the result of the keyword from the indicies
+   */
+  getResult(keyword, fileName) {
+    const searchResult = {},
+      keywords = InvertedIndex.cleanValues(keyword),
+      fileIndex = this.indicies[fileName],
+      currentToken = Object.keys(this.indicies[fileName]);
+    if (keywords.length === 0) {
+      throw new Error('Search for Aplhanumeric values only not special character');
+    }
+    keywords.forEach((elem) => {
+      if (currentToken.includes(elem)) {
+        searchResult[elem] = fileIndex[elem];
+      } else {
+        searchResult[elem] = [];
+      }
+    });
+    return searchResult;
+  }
+
+  /**
+   * @cleanValues method
+   * @param {str} str
```

Comment:
What is _**str**_? Please be descriptive with your naming. Abbreviations are not allowed.

How is _**str**_ a dataType?